sound/hda/ext/hdac_ext_stream.c

```diff
@@ -350,7 +387,7 @@ struct hdac_ext_stream *snd_hdac_ext_stream_assign(struct hdac_bus *bus,
 		return hstream;
 
 	case HDAC_EXT_STREAM_TYPE_HOST:
-		return hdac_ext_host_stream_assign(bus, substream);
+		return hdac_ext_host_stream_pcm_assign(bus, substream);
```

Comment:
@crojewsk why not call hdac_ext_host_stream_assign with substream->stream directly here? Do we really need hdac_ext_host_stream_pcm_assign() at all?

Reply:
You still need to assign the res->hstream.(substream|cstream). If we go with "direction", then each invocation has to actually assign the stream explicitly. Existing solution is cohesive and reduces the duplication in the long run.